src/plugins/navigation/public/top_nav_menu/top_nav_menu.tsx

```diff
@@ -73,14 +85,24 @@ export function TopNavMenu<QT extends AggregateQuery | Query = Query>(
     if (!badges || badges.length === 0) return null;
     return (
       <EuiBadgeGroup className={'kbnTopNavMenu__badgeGroup'}>
-        {badges.map((badge: EuiBadgeProps & { badgeText: string }, i: number) => {
-          const { badgeText, ...badgeProps } = badge;
-          return (
-            <EuiBadge key={`nav-menu-badge-${i}`} {...badgeProps}>
-              {badgeText}
-            </EuiBadge>
-          );
-        })}
+        {badges.map(
+          (
+            badge: EuiBadgeProps & {
+              badgeText: string;
+              toolTipProps: EuiToolTipProps;
+            },
+            i: number
+          ) => {
+            const { badgeText, toolTipProps, ...badgeProps } = badge;
+            return (
+              <EuiToolTip {...toolTipProps}>
+                <EuiBadge key={`nav-menu-badge-${i}`} {...badgeProps}>
+                  {badgeText}
+                </EuiBadge>
+              </EuiToolTip>
```

Comment:
**nit:** Makes me nervous to always wrap every badge with an `EuiTooltip`, since we're not the only consumer of the `TopNavMenu` component.... I don't **think** any other implementation uses badges (searched quickly and couldn't find any), but I wonder if we want to make the `toolTipProps` optional and conditionally wrap it?

Reply:
I also checked for any other consumers that use badges, but didn't see any. But I agree that tooltips could be optional for future consumers. I'm a little surprised `tooltip` is not already an optional prop on `EuiBadge` 🤷 . Updated in [fb3e1a0](https://github.com/elastic/kibana/pull/154253/commits/fb3e1a015e1e4b1734f4b74f162703ea2711283e).